crates/oxc_transformer/src/proposals/explicit_resource_management.rs

```diff
@@ -210,11 +210,61 @@ impl<'a> Traverse<'a> for ExplicitResourceManagement<'a, '_> {
         if let Some((new_stmts, needs_await, using_ctx)) =
             self.transform_statements(&mut body.statements, ctx.current_hoist_scope_id(), ctx)
         {
-            // FIXME: this creates the scopes in the correct place, however we never move the bindings contained
-            // within `new_stmts` to the new scope.
             let block_stmt_scope_id =
                 ctx.insert_scope_below_statements(&new_stmts, ScopeFlags::empty());
 
+            // TODO: make this less hacky
+            for stmt in &new_stmts {
+                match stmt {
```

Comment:
i really hate how i've done this, but it works for now.

why does `ctx.insert_scope_below_statements` not move the scopes of the symbols it sees? this seems like something obvious it should do (although there's probably a great reason why it doesn't), because i can't think of a situation where you would be inserting a scope but not moving the symbols scopes inside it

Reply:
Can you use `Scoping` data instead? Loop through the bindings of the scope and move any symbols which are block scoped.

But I may well be wrong and that doesn't work.

Unfortunately may also need to deal with `BindingIdentifier`s in `TS*` nodes e.g. `TSEnumDeclaration`. TS transform will remove them anyway, but if this transform moves those nodes first, the bindings need to be moved too, to avoid confusing the TS transform.

Reply:
Ah feck I think I am wrong. Presumably that wouldn't work if it's a `Function`, because of bindings for function params. Or maybe it's OK if function params aren't block scoped?

Reply:
apologies. i thought about replying, but forgot to!

Yes it's exactly that reason.

in this example, we've gone from:
```ts
function f(arg0) {
    using test = arg0();
}
```

to
```ts
function f(arg0) {
  try {
    var _usingCtx = babelHelpers.usingCtx();
    const x = _usingCtx.u(arg0());
  } catch (_) {
    _usingCtx.e = _;
  } finally {
    _usingCtx.d();
  }
}
```

so only the bindings inside the function body (note `arg0` is in the scope but shouldn't be moved), should be moved.

My thought's are that id's be *super* helpful if `ChildScopeCollector` also returned binding ids as well as wel as the socpes that it moved. but may have to flesh it out a bit more

Reply:
Actually, looking at `ChildScopeCollector` is used currently. it is never used in a scenario where the scope or bindings would change. it feels reasonable to adjust it as described above.